widgets/net_freebsd.lua

```diff
@@ -0,0 +1,85 @@
+-- {{{ Grab environment
+local tonumber = tonumber
+local os = { time = os.time }
+local setmetatable = setmetatable
+local helpers = require("vicious.helpers")
+-- }}}
+
+
+-- Net: provides state and usage statistics of all network interfaces
+-- vicious.widgets.net
+local net_freebsd = {}
+
+
+-- Initialize function tables
+local nets = {}
+-- Variable definitions
+local unit = { ["b"] = 1, ["kb"] = 1024,
+    ["mb"] = 1024^2, ["gb"] = 1024^3
+}
+
+-- {{{ Net widget type
+local function worker(format, warg)
+    if not warg then return end
+
+    local args = {}
+    local buffer = nil
+    local f = io.popen("netstat -n -b -I " .. helpers.shellquote(warg))
+    local now = os.time()
+    
+    for line in f:lines() do
+        if line:find("<Link") or line:find("Name ") then -- skipping missleading lines
+            goto continue
+        end
+
+        local split = { line:match(("([^%s]*)%s*"):rep(12)) }
+
+        if buffer == nil then
+            buffer = { tonumber(split[8]), tonumber(split[11]) } -- recv (field 8) and send (field 11)
+        else
+            buffer = { buffer[1] + tonumber(split[8]), buffer[2] + tonumber(split[11]) }
+        end
+
+        ::continue::
```

Comment:
this would break lua 5.1, for the moment I would stick to:

```
for line in f:lines() do
      if not (line:find("<Link") and line:find("Name ")) then -- skipping missleading lines
          # ...
      end
end 
```